### PR TITLE
sort items by price implemented

### PR DESCRIPTION
**Pull request by kaelinpsalazar**
Implemented testing, model, routes, and controller changes for sort by price

### CODE REVIEW

**Review by Sgalvin36 — APPROVED** [review]
Took the advice of Abdul, simplified our tests to test both the presence of a key as well as its type in a single test instead of two tests. Looks good with the if/else logic used for parameters.

File: spec/requests/v1/items_request_spec.rb [hunk 1]
```diff
@@ -38,20 +38,41 @@
...
         items = JSON.parse(response.body, symbolize_names: true)
 
+        
+
         items[:data].each do |item|
-            expect(item[:id].to_i).to be_an(Integer)
+            expect(item[:id]).to be_an(String)
+            # check for type
 
-            expect(item[:attributes]).to have_key(:name)
```
Comment (kylomite) [inline spec/requests/v1/items_request_spec.rb:44]: Taking Adbul's advice here. the fact that we are checking the values of the keys within `attribute` makes the `.to have_key` redundant, I will implement this for my unit tests for merchants as well.

File: spec/requests/v1/items_request_spec.rb [hunk 2]
```diff
@@ -38,20 +38,41 @@
...
+        get '/api/v1/items?sort=unit_price'
+
+        expect(response).to be_successful
+
+        items = JSON.parse(response.body, symbolize_names: true)
+
+        prices = items[:data].map { |item| item[:attributes][:unit_price] }
+        expect(prices).to eq(prices.sort)
+
+        items[:data].each do |item|
```
Comment (kylomite) [inline spec/requests/v1/items_request_spec.rb:65]: I am curious about this part of the test? Is there an expectation to check this on every test? This it block is testing that the items are sorted when the query param is in place. This is checked in line 63, but this bit down here is testing for something else. Its not bad to have thorough testing but i'm curious about what the justification is in this context.

**Review by kylomite — APPROVED** [review]
Good implementation, it seems like the behavior will run as expected, but  I do have some questions for my own understanding. I wrote individual comments on the  commit, but I will bring this up when we have work time as well.

**Pull request merged**